<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Generate Starting Trees For Combined Molecular, Morphological and Stratigraphic Data • treestartr</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Generate Starting Trees For Combined Molecular, Morphological and Stratigraphic Data">
<meta property="og:description" content="Combine a list of taxa with a phylogeny to generate a starting tree for use in
    total evidence dating analyses.">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">treestartr</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    

    
    
<p><img src="images/logo1.png" alt="logo"><a href="https://travis-ci.org/wrightaprilm/treeStartR"><img src="https://travis-ci.org/ropensci/treeStartR.svg?branch=master" alt="Build Status"></a> <a href="https://codecov.io/gh/ropensci/treeStartR"><img src="https://codecov.io/gh/ropensci/treeStartR/branch/master/graph/badge.svg" alt="codecov"></a> <a href="https://github.com/ropensci/onboarding/issues/239"><img src="https://badges.ropensci.org/239_status.svg"></a></p>
<div id="treestartr" class="section level2">
<h2 class="hasAnchor">
<a href="#treestartr" class="anchor"></a>treeStartR</h2>
<p>treeStartR is an R package for automated generation of starting trees for total-evidence phylogenetic analyses.</p>
<p>Phylogenetic trees, and particularly time-scaled phylogenetic trees, are increasingly estimated using parameter-rich models of evolution <span class="citation">(Lartillot, Lepage, and Blanquart 2009; Wright, Lloyd, and Hillis 2016)</span> and models incorporating macroevoloutionary processes <span class="citation">(Heath, Huelsenbeck, and Stadler 2014)</span>. Finding a starting tree with a computable likelihood to perform a Bayesian MCMC under these complex models can be a challenge, particularly when when estimation involves many taxa, large datasets, and missing data.</p>
<p>Different phylogenetic estimation packages allow users to find starting trees in different ways, such as estimating a tree under parsimony (<span class="citation">Stamatakis (2014)</span>) or neighbor-joining <span class="citation">(Bouckaert et al. 2014)</span>, randomly adding taxa to the tree <span class="citation">(Bouckaert et al. 2014; Stamatakis 2014)</span>, or allowing the user to specify the tree. Addition of taxa is usually performed based on data - that is, using an algorithm (such as parsimony or neighbor-joining) to add the tips to the tree. However, in analyses of the fossil record, specimens may be included for which there are no molecular or morphological characters available, but the taxonomy of the specimen is known via expert opinion. This is the case with many specimens harvested via repositories such as the Paleobiology Database.</p>
<p>The purpose of this package is to allow users to efficiently add taxa to a given tree to generate a reasonable starting tree. Functions in this package allow taxa to be added to a tree according to either their taxonomy (if other specimens from the same genus are present) in function <code>present_tippr</code>, at random (<code>rand_absent_tippr</code>), or via other user-specified groupings (<code>text_placr</code>). The package uses functions from phytools <span class="citation">(Revell 2012)</span> and ape <span class="citation">(Paradis, Claude, and Strimmer 2004)</span>.</p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>Install the development version directly from Github</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">library</span>(devtools)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/install_github">install_github</a></span>(<span class="st">"wrightaprilm/treeStartR"</span>)</a></code></pre></div>
</div>
<div id="usage-example" class="section level2">
<h2 class="hasAnchor">
<a href="#usage-example" class="anchor"></a>Usage Example</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw">library</span>(treestartr)</a></code></pre></div>
<pre><code>## Loading required package: phytools</code></pre>
<pre><code>## Loading required package: ape</code></pre>
<pre><code>## Loading required package: maps</code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw">data</span>(bears)</a></code></pre></div>
<p>First, we need to load a list of the total set of taxa present in the tree. The “total set” refers to any taxa that will be included in your analysis. This can be either a CSV or a TSV file. A sample list, bears, has been provided as part of the bears data object, but you can also generate one using the function dataf_parser. Note that if you have higher order taxa, for example, if you are placing a fossil that you have identified to family level, but not species level, these should be included as “family_spp.” treeStartR relies on the underscore to separate between the higher order and lower order taxa. Use consistent formatting between the tree and taxon list - i.e, do not call a taxon “Ge_spp” on the tree and “Genus_spp” in the taxon list.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">tax_frame &lt;-<span class="st"> </span><span class="kw"><a href="reference/dataf_parsr.html">dataf_parsr</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/system.file">system.file</a></span>(<span class="st">"extdata/bears_taxa.tsv"</span>, <span class="dt">package =</span> <span class="st">"treestartr"</span>))</a></code></pre></div>
<pre><code>## File format is appropriate</code></pre>
<pre><code>## .tsv file ending detected.</code></pre>
<pre><code>## File processing complete.</code></pre>
<pre><code>## Taxon dataframe names formatted correctly</code></pre>
<p>Next, we find out which of the taxa from our total set are not represented on the tree already. This function takes as input a tree, with or without branch lengths, but without annotations (such as 95% HPDs). It also takes the total set of taxa generated by <code>dataf_parser</code>:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">absent_list &lt;-<span class="st"> </span><span class="kw"><a href="reference/genera_strippr.html">genera_strippr</a></span>(tree, tax_frame)</a></code></pre></div>
</div>
<div id="adding-tips-with-congeners" class="section level2">
<h2 class="hasAnchor">
<a href="#adding-tips-with-congeners" class="anchor"></a>Adding tips with congeners</h2>
<p>Finally, we add the tips that are not present to the tree. If there are other representatives of the same genera as an absent taxon (for example, adding an additional “Ursus” species to the example tree), those taxa will be used to place the tip. If there are multiple species of the genera, the new tip will subtend the most recent common ancestor of the tips already on the tree. If there is only one representative, the tip will subtend the parent node of that taxon.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">new_tree &lt;-<span class="st"> </span><span class="kw"><a href="reference/present_tippr.html">present_tippr</a></span>(tree, absent_list)</a></code></pre></div>
<pre><code>## Tree tip names formatted correctly</code></pre>
<pre><code>## [1] "Indarctos_arctoides" "Indarctos_vireti"</code></pre>
<pre><code>## Adding tip Indarctos_punjabiensis</code></pre>
<pre><code>## [1] "Ursus_americanus" "Ursus_thibetanus" "Ursus_arctos"    
## [4] "Ursus_maritimus"  "Ursus_spelaeus"</code></pre>
<pre><code>## Adding tip Ursus_abstrusus</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="kw">plot</span>(new_tree)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<p>After we have added our tips, we can check how many tips remain to be added, and which they are:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1"><span class="kw"><a href="reference/genera_strippr.html">genera_strippr</a></span>(new_tree, tax_frame)</a></code></pre></div>
<pre><code>## [1] "Ailurarctos_lufengensis" "Kretzoiarctos_beatrix"  
## [3] "Parictis_montanus"       "Ursavus_brevirhinus"    
## [5] "Ursavus_primaevus"</code></pre>
</div>
<div id="adding-tips-manually" class="section level2">
<h2 class="hasAnchor">
<a href="#adding-tips-manually" class="anchor"></a>Adding tips manually</h2>
<p>We can also add the tips that have no congeners. This function will ask for input. A pop-up will be produced, showing node labels. When the program asks for input, you will tell it what node you would like the tip to subtend. Alternatively, if you would like to place the tip as sister to a tip on the tree, enter the number of the tip. Tip numbers are highlighted in yellow.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1">new_tree &lt;-<span class="st"> </span><span class="kw"><a href="reference/absent_tippr.html">absent_tippr</a></span>(tree, absent_list)</a></code></pre></div>
</div>
<div id="adding-tips-at-random" class="section level2">
<h2 class="hasAnchor">
<a href="#adding-tips-at-random" class="anchor"></a>Adding tips at random</h2>
<p>Or, if there are no congeners, you may choose to add tips at random:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1">new_tree &lt;-<span class="st"> </span><span class="kw"><a href="reference/rand_absent_tippr.html">rand_absent_tippr</a></span>(tree, absent_list)</a></code></pre></div>
<pre><code>## Adding tips at random node: Ailurarctos_lufengensis</code></pre>
<pre><code>## [1] 25</code></pre>
<pre><code>## Adding tips at random node: Kretzoiarctos_beatrix</code></pre>
<pre><code>## [1] 4</code></pre>
<pre><code>## Adding tips at random node: Parictis_montanus</code></pre>
<pre><code>## [1] 19</code></pre>
<pre><code>## Adding tips at random node: Ursavus_brevirhinus</code></pre>
<pre><code>## node is the root. returning NULL
## [1] 11</code></pre>
<pre><code>## Adding tips at random node: Ursavus_primaevus</code></pre>
<pre><code>## [1] 3</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" data-line-number="1"><span class="kw">plot</span>(new_tree)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<p>Running the command again to generate a second random addition tree, and comparing it to the first using the RF distance <span class="citation">(Robinson and Foulds 1981)</span>, shows us that the tree topologies truly are different.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb35-1" data-line-number="1">second_random &lt;-<span class="st"> </span><span class="kw"><a href="reference/rand_absent_tippr.html">rand_absent_tippr</a></span>(tree, absent_list)</a></code></pre></div>
<pre><code>## Adding tips at random node: Ailurarctos_lufengensis</code></pre>
<pre><code>## [1] 26</code></pre>
<pre><code>## Adding tips at random node: Kretzoiarctos_beatrix</code></pre>
<pre><code>## [1] 8</code></pre>
<pre><code>## Adding tips at random node: Parictis_montanus</code></pre>
<pre><code>## [1] 36</code></pre>
<pre><code>## Adding tips at random node: Ursavus_brevirhinus</code></pre>
<pre><code>## [1] 28</code></pre>
<pre><code>## Adding tips at random node: Ursavus_primaevus</code></pre>
<pre><code>## [1] 27</code></pre>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb46-1" data-line-number="1">phangorn<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/phangorn/topics/treedist">RF.dist</a></span>(new_tree, second_random)</a></code></pre></div>
<pre><code>## Trees are not binary!</code></pre>
<pre><code>## [1] 23</code></pre>
</div>
<div id="adding-tips-via-csv" class="section level2">
<h2 class="hasAnchor">
<a href="#adding-tips-via-csv" class="anchor"></a>Adding tips via CSV</h2>
<p>Lastly, you may have a TSV file that specifies the tips to be added, and a taxon set. treeStartR will locate the MRCA of the taxon set, and add the tips subtending that node.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb49-1" data-line-number="1">new_tree &lt;-<span class="st"> </span><span class="kw"><a href="reference/text_placr.html">text_placr</a></span>(tree, mrca_df)</a></code></pre></div>
<pre><code>## Placing tip Kretzoiarctos_beatrix</code></pre>
<pre><code>## via relatives Indarctos_arctoides, Indarctos_vireti</code></pre>
<pre><code>##  at node 25</code></pre>
<pre><code>## Placing tip Ursus_abstrusus</code></pre>
<pre><code>## via relatives Ursus_arctos, Ursus_spelaeus, Ursus_americanus</code></pre>
<pre><code>##  at node 30</code></pre>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb56-1" data-line-number="1"><span class="kw">plot</span>(new_tree)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
</div>
<div id="resolving-polytomies" class="section level2">
<h2 class="hasAnchor">
<a href="#resolving-polytomies" class="anchor"></a>Resolving polytomies</h2>
<p>By default, treeStartR outputs trees with polytomies. This is because the placement of tips is not being estimated from data, but rather placed arbitrarily. RevBayes and BEAST2 read non-bifurcating trees. If you are working with analytical software that does not, you may want to resolve polytomies before export, such as with ape’s <code>multi2di</code> <a href="https://www.rdocumentation.org/packages/ape/versions/5.1/topics/multi2di">function</a>:</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb57-1" data-line-number="1">ape<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/ape/topics/multi2di">multi2di</a></span>(new_tree)</a></code></pre></div>
<pre><code>## 
## Phylogenetic tree with 17 tips and 21 internal nodes.
## 
## Tip labels:
##  Ailuropoda_melanoleuca, Agriarctos_spp, Kretzoiarctos_beatrix, Indarctos_arctoides, Indarctos_vireti, Ursus_abstrusus, ...
## 
## Rooted; no branch lengths.</code></pre>
</div>
<div id="subtrees-and-clade-constraints" class="section level2">
<h2 class="hasAnchor">
<a href="#subtrees-and-clade-constraints" class="anchor"></a>Subtrees and Clade Constraints</h2>
<p>All tip addition functions, except <code>absent_tippr</code> have the capability to write out <a href="https://revbayes.github.io/tutorials/fbd/#Exercise-ModelFBD">RevBayes-formatted</a> clade constraint strings. The <code>echo_revbayes</code> argument allows this.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb59-1" data-line-number="1"><span class="kw"><a href="reference/text_placr.html">text_placr</a></span>(tree, mrca_df, <span class="dt">echo_revbayes =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<pre><code>## Placing tip Kretzoiarctos_beatrix</code></pre>
<pre><code>## via relatives Indarctos_arctoides, Indarctos_vireti</code></pre>
<pre><code>##  at node 25</code></pre>
<pre><code>## clade( "Indarctos_arctoides", "Indarctos_vireti", "Kretzoiarctos_beatrix" )</code></pre>
<pre><code>## Placing tip Ursus_abstrusus</code></pre>
<pre><code>## via relatives Ursus_arctos, Ursus_spelaeus, Ursus_americanus</code></pre>
<pre><code>##  at node 30</code></pre>
<pre><code>## clade( "Ursus_arctos", "Ursus_spelaeus", "Ursus_americanus", "Ursus_abstrusus" )</code></pre>
<pre><code>## 
## Phylogenetic tree with 17 tips and 19 internal nodes.
## 
## Tip labels:
##  Ailuropoda_melanoleuca, Agriarctos_spp, Kretzoiarctos_beatrix, Indarctos_arctoides, Indarctos_vireti, Ursus_abstrusus, ...
## 
## Rooted; no branch lengths.</code></pre>
<p>Likewise, subtrees with tips added can be printed to the screen.</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb69-1" data-line-number="1"><span class="kw"><a href="reference/present_tippr.html">present_tippr</a></span>(tree, absent_list, <span class="dt">echo_subtrees =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<pre><code>## Tree tip names formatted correctly</code></pre>
<pre><code>## Echoing Subtrees to Screen</code></pre>
<pre><code>## [1] "Indarctos_arctoides" "Indarctos_vireti"</code></pre>
<pre><code>## Adding tip Indarctos_punjabiensis</code></pre>
<pre><code>## Subtree:  (Indarctos_punjabiensis,Indarctos_arctoides,Indarctos_vireti);[1] "Ursus_americanus" "Ursus_thibetanus" "Ursus_arctos"    
## [4] "Ursus_maritimus"  "Ursus_spelaeus"</code></pre>
<pre><code>## Adding tip Ursus_abstrusus</code></pre>
<pre><code>## Subtree:  (Ursus_abstrusus,((Ursus_americanus,Ursus_thibetanus),Helarctos_malayanus),((Ursus_arctos,Ursus_maritimus),Ursus_spelaeus));</code></pre>
<pre><code>## 
## Phylogenetic tree with 17 tips and 19 internal nodes.
## 
## Tip labels:
##  Ailuropoda_melanoleuca, Agriarctos_spp, Indarctos_punjabiensis, Indarctos_arctoides, Indarctos_vireti, Ursus_abstrusus, ...
## 
## Rooted; no branch lengths.</code></pre>
</div>
<div id="outputting-results" class="section level2">
<h2 class="hasAnchor">
<a href="#outputting-results" class="anchor"></a>Outputting results</h2>
<p>The final tree can be output using standard functions in ape, such as the <code>write.nexus()</code> <a href="https://www.rdocumentation.org/packages/ape/versions/5.2/topics/write.tree">function</a>:</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb78-1" data-line-number="1">ape<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/ape/topics/write.nexus">write.nexus</a></span>(new_tree, <span class="dt">file =</span> <span class="st">"data/export_example.tre"</span>)</a></code></pre></div>
</div>
<div id="flowchart" class="section level2">
<h2 class="hasAnchor">
<a href="#flowchart" class="anchor"></a>Flowchart</h2>
<p>Here is the flowchart of the functions in <code>treeStartR</code>, and when to use them.</p>
<p><img src="images/flowchart/flowchart.001.png"><a href="https://ropensci.org"><img src="https://ropensci.org/public_images/ropensci_footer.png" alt="ropensci_footer"></a></p>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<div id="refs" class="references">
<div id="ref-beast2">
<p>Bouckaert, Remco, Joseph Heled, Denise Kühnert, Tim Vaughan, Chieh-Hsi Wu, Dong Xie, Marc A. Suchard, Andrew Rambaut, and Alexei J. Drummond. 2014. “BEAST 2: A Software Platform for Bayesian Evolutionary Analysis.” <em>PLOS Computational Biology</em> 10 (4). Public Library of Science: 1–6. <a href="https://doi.org/10.1371/journal.pcbi.1003537" class="uri">https://doi.org/10.1371/journal.pcbi.1003537</a>.</p>
</div>
<div id="ref-fbd">
<p>Heath, Tracy A, John P Huelsenbeck, and Tanja Stadler. 2014. “The fossilized birth-death process for coherent calibration of divergence-time estimates.” <em>Proceedings of the National Academy of Sciences</em> 111 (29). National Acad Sciences: E2957–E2966.</p>
</div>
<div id="ref-catmodel">
<p>Lartillot, N., T. Lepage, and S. Blanquart. 2009. “PhyloBayes 3: A Bayesian Software Package for Phylogenetic Reconstruction and Molecular Dating.” <em>Bioinformatics</em> 25 (17). Oxford Univ Press: 2286.</p>
</div>
<div id="ref-ape">
<p>Paradis, E., J. Claude, and K. Strimmer. 2004. “APE: Analyses of Phylogenetics and Evolution in R Language.” <em>Bioinformatics</em> 20 (2). Oxford Univ Press: 289–90.</p>
</div>
<div id="ref-phytools">
<p>Revell, Liam J. 2012. “Phytools: An R Package for Phylogenetic Comparative Biology (and Other Things).” <em>Methods in Ecology and Evolution</em> 3 (2): 217–23. <a href="https://doi.org/10.1111/j.2041-210X.2011.00169.x" class="uri">https://doi.org/10.1111/j.2041-210X.2011.00169.x</a>.</p>
</div>
<div id="ref-Robinson1981">
<p>Robinson, DF, and LR Foulds. 1981. “Comparison of phylogenetic trees.” <em>Math. Biosci</em> 53 (1-2): 131–47.</p>
</div>
<div id="ref-raxml8">
<p>Stamatakis, Alexandros. 2014. “RAxML Version 8: A Tool for Phylogenetic Analysis and Post-Analysis of Large Phylogenies.” <em>Bioinformatics</em> 30 (9): 1312–3. <a href="https://doi.org/10.1093/bioinformatics/btu033" class="uri">https://doi.org/10.1093/bioinformatics/btu033</a>.</p>
</div>
<div id="ref-wright2016">
<p>Wright, April M., Graeme T. Lloyd, and David M. Hillis. 2016. “Modeling Character Change Heterogeneity in Phylogenetic Analyses of Morphology Through the Use of Priors.” <em>Systematic Biology</em> 65 (4): 602–11.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Report a bug at <br><a href="https://github.com/wrightaprilm/treeStartR/issues">https://​github.com/​wrightaprilm/​treeStartR/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>April Wright <br><small class="roles"> Author, maintainer </small>  </li>
</ul>
</div>

      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by April Wright.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
